### PR TITLE
docs: document usage as library (#871)

### PR DESCRIPTION
**Pull request by miparnisari**
Follow-up of https://github.com/openfga/openfga/issues/871

### CODE REVIEW

File: pkg/server/server_test.go [hunk 1]
```diff
@@ -45,6 +45,66 @@ func init() {
 	}
 }
 
+func ExampleNewServerWithOpts() {
+	openfga, err := NewServerWithOpts(WithDatastore(memory.New()))
```
Comment (adriantam) [inline pkg/server/server_test.go:49]: The interesting part is here, should we document what options that are typical to be used as library?  Also, do we need to note what sort of middleware we expect user to setup?

Comment (miparnisari) [reply pkg/server/server_test.go:49]: > should we document what options that are typical to be used as library

I'm not sure if this is the best place to put that, i don't want to be too verbose, or to repeat documentation. We could point to https://github.com/openfga/openfga/blob/c461c42a3993af58fa2b4c60242ef97cb7765d09/.config-schema.json? I'm open to ideas

> need to note what sort of middleware we expect user to setup?

🤔 You don't need middleware to use it as a library. At one point in the past we realized that if you used it as a library you had to add the gRPC validation middleware, but that is not the case anymore.

File: README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ OpenFGA is designed to make it easy for developers to model their application pe
...
 
-It offers an [HTTP API](https://openfga.dev/api/service) and a [gRPC API](https://buf.build/openfga/api/file/main:openfga/v1/openfga_service.proto). It has SDKs for [Node.js/JavaScript](https://www.npmjs.com/package/@openfga/sdk), [GoLang](https://github.com/openfga/go-sdk), [Python](https://github.com/openfga/python-sdk) and [.NET](https://www.nuget.org/packages/OpenFga.Sdk). Look in our [Community section](https://github.com/openfga/community#community-projects) for third-party SDKs and tools.
+It offers an [HTTP API](https://openfga.dev/api/service) and a [gRPC API](https://buf.build/openfga/api/file/main:openfga/v1/openfga_service.proto). It has SDKs for [Java](https://central.sonatype.com/artifact/dev.openfga/openfga-sdk), [Node.js/JavaScript](https://www.npmjs.com/package/@openfga/sdk), [GoLang](https://github.com/openfga/go-sdk), [Python](https://github.com/openfga/python-sdk) and [.NET](https://www.nuget.org/packages/OpenFga.Sdk). Look in our [Community section](https://github.com/openfga/community#community-projects) for third-party SDKs and tools. It can also be used as a library (see [example](https://pkg.go.dev/github.com/openfga/openfga/pkg/server#pkg-examples)).
```
Comment (aaguiarz) [inline README.md:22]: That 'example' does not look like an example :) but a documentation page for the Server 

Should we have an example somewhere?

Comment (miparnisari) [reply README.md:22]: Once we merge this PR that link will be automatically updated to include the example I posted here. Similar to https://pkg.go.dev/github.com/openfga/openfga@v1.3.7/pkg/storage#pkg-examples

File: pkg/server/server_test.go (hunk 1 above)
Comment (jon-whit) [reply pkg/server/server_test.go:49]: The godoc should serve as the list 😄 . That's the point.

Focus on good godoc and we'll get good documentation on go.pkg.dev

Comment (miparnisari) [reply pkg/server/server_test.go:49]: Are we good to merge as is then?

**Comment by rhamzeh** [comment]
@miparnisari  can we add docuumentation in how to 
* specify the pg store
* override the default config
* preload the in mem store with data?

**Review by rhamzeh — APPROVED** [review]

**Review by willvedd — APPROVED** [review]

**Pull request merged**